haskell-src/Concordium/Types/Parameters.hs

```diff
@@ -1582,6 +1600,59 @@ instance HashableTo Hash.Hash TimeoutParameters where
 
 instance (Monad m) => MHashableTo m Hash.Hash TimeoutParameters
 
+-- * Finalization committee parameters for consensus v1.
+
+-- |Finalization committee parameters
+-- These parameters control who and how many bakers will be
+-- eligible for joinin the finalization committee.
+data FinalizationCommitteeParameters = FinalizationCommitteeParameters
+    { -- |Minimum number of bakers to include in the finalization committee.
+      _fcpMinFinalizers :: !Word32,
+      -- |Maximum number of bakers to include in the finalization committee.
+      _fcpMaxFinalizers :: !Word32,
+      -- |Determining the staking threshold required for being eligible the finalization committee.
+      -- The required amount is given by @total staked ccd / _fcpFinalizerRelativeStakeThreshold@
+      _fcpFinalizerRelativeStakeThreshold :: !Word32
+    }
+    deriving (Eq, Show)
+
+makeLenses ''FinalizationCommitteeParameters
+
+instance Serialize FinalizationCommitteeParameters where
+    put FinalizationCommitteeParameters{..} = do
+        put _fcpMinFinalizers
+        put _fcpMaxFinalizers
+        put _fcpFinalizerRelativeStakeThreshold
+    get = do
+        _fcpMinFinalizers <- get
+        unless (_fcpMinFinalizers > 0) $ fail "the minimum number of finalizers must be positive."
+        _fcpMaxFinalizers <- get
+        unless (_fcpMaxFinalizers > _fcpMinFinalizers) $ fail "The maximum number of finalizers must be greater than minimumFinalizers."
+        _fcpFinalizerRelativeStakeThreshold <- get
+        return FinalizationCommitteeParameters{..}
+
+instance HashableTo Hash.Hash FinalizationCommitteeParameters where
+    getHash = Hash.hash . encode
+
+instance (Monad m) => MHashableTo m Hash.Hash FinalizationCommitteeParameters
+
+instance ToJSON FinalizationCommitteeParameters where
+    toJSON FinalizationCommitteeParameters{..} =
+        object
+            [ "maximumFinalizers" AE..= _fcpMinFinalizers,
+              "minimumFinalizers" AE..= _fcpMaxFinalizers,
+              "finalizerRelativeStakeThreshold" AE..= _fcpFinalizerRelativeStakeThreshold
+            ]
+
+instance FromJSON FinalizationCommitteeParameters where
+    parseJSON = withObject "FinalizationCommitteeParameters" $ \o -> do
+        _fcpMinFinalizers <- o .: "minimumFinalizers"
+        unless (_fcpMinFinalizers > 0) $ fail "the minimum number of finalizers must be positive."
+        _fcpMaxFinalizers <- o .: "maximumFinalizers"
+        unless (_fcpMaxFinalizers > _fcpMinFinalizers) $ fail "The maximum number of finalizers must be greater than minimumFinalizers."
+        _fcpFinalizerRelativeStakeThreshold <- o .: "finalizerRelativeStakeThreshold"
```

Comment:
Shouldn't we here also check that `_fcpFinalizerRelativeStakeThreshold` is positive?

Reply:
It's a word32. and now also a `PartsOfOverOneHundredThousands` which checks that the value supplied is within. [0,1]

Reply:
Then why do we check here that `_fcpMinFinalizers > 0`?

Reply:
Do we want to allow 0 finalizers? Its not a natural number, but an unsigned integer

Reply:
Yes, and Max to 10! Off to the races…

Reply:
If you are inclined to be pedantic, why is it not allowed that min = max?

But it likely does not matter since this FromJSON instance isn't going to be used

Reply:
Now allowing min = max.